### PR TITLE
[MM-49373] Bucket and Helm deletion fixes

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary

- Use `HeadBucket` instead of `GetBucketPolicy` to check if a bucket exists or not.
- Check chart is present before trying to delete it.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-49373

https://mattermost.atlassian.net/browse/MM-49377

#### Release Note

```release-note
None
```

### CODE REVIEW

**Review by stafot — APPROVED** [review]

File: internal/provisioner/nginx_internal.go [hunk 1]
```diff
@@ -95,15 +95,10 @@ func (n *nginxInternal) ActualVersion() *model.HelmUtilityVersion {
...
-	helm := newHelmDeployment(
-		chartNameNginxInternal,
-		chartDeploymentNameNginxInternal,
-		namespaceNginxInternal,
-		n.kubeconfigPath,
-		nil,
-		defaultHelmDeploymentSetArgument,
-		n.logger,
-	)
+	helm, err := n.NewHelmDeployment()
```
Comment (mirshahriar) [inline internal/provisioner/nginx_internal.go:98]: Why are we adding this? `NewHelmDeployment` does some extra tasks. I think we do not need that to delete.

**Review by mirshahriar — APPROVED** [review]

File: internal/provisioner/helm_utils.go [hunk 2]
```diff
@@ -68,14 +68,38 @@ func (d *helmDeployment) Update() error {
...
 	logger := d.logger.WithField("helm-delete", d.chartDeploymentName)
 
-	logger.Infof("Deleting helm chart %s", d.chartDeploymentName)
-	err := deleteHelmChart(*d, d.kubeconfigPath, logger)
+	// Ensure the chart is present before deletion
+	exists, err := d.Exists()
 	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("got an error trying to delete the helm chart %s", d.chartDeploymentName))
+		return err
+	} else if !exists {
```
Comment (mirshahriar) [inline internal/provisioner/helm_utils.go:75]:
```suggestion
	}
        if !exists {
```

**Review by mirshahriar — APPROVED** [review]

File: internal/provisioner/helm_utils.go (hunk 2 above)
Comment (mirshahriar) [reply internal/provisioner/helm_utils.go:75]: nit: we do not need `else` here

Comment (fmartingr) [reply internal/provisioner/helm_utils.go]: I don't know what I was thinking 😂

File: internal/provisioner/nginx_internal.go (hunk 1 above)
Comment (fmartingr) [reply internal/provisioner/nginx_internal.go:98]: I wanted to be sure that we use the same helm deployment in both creation and deletion, this was the only provisioner that used a different one in the two methods.

Comment (fmartingr) [reply internal/provisioner/nginx_internal.go]: @mirshahriar Take a look at the changes

**Pull request merged**